### ISSUE TITLE
static link to version 0.5.1 of the micro patcher

### ISSUE THREAD

**Issue by Ausdauersportler** [moosethegoose2213/automator-for-barrykn-micropatcher#6]
Hi! 

Really cool development, I was just inspecting the instructions @jackluke made to make such an enhancement myself when I found your code and tool. Would it be possible that you link in the micro patcher source code like a fork instead of creating a complete copy. 

Background: The micro patcher has more that 80 forks and some of them may step forward and it would be nice to use your tool with the upcoming versions. 

Thanks in advance

**Comment by moosethegoose2213** [comment]
An option would be to replace the folder that holds the Micropatcher with the fork, and as long as the names are the same it'll execute that version instead. In a future version it will be replaced with a compressed version, so you would need to replace the compressed file.

**Comment by minh-ton** [comment]
> An option would be to replace the folder that holds the Micropatcher with the fork, and as long as the names are the same it'll execute that version instead. In a future version it will be replaced with a compressed version, so you would need to replace the compressed file.

@moosethegoose2213 How about adding a "Custom micropatcher fork" feature instead of that.... 🤔

**Comment by Ausdauersportler** [comment]
Thanks for coming back! I did this already and put my version 0.5.4 including some OpenCore add one into it which will only be used with iMac 2009-2011. Everything else remains the same except I added the Night Shift patch and an HDMI audio patch for MacPro3,1. There are only three forks off your current project. You will find it easily :-)

**Comment by minh-ton** [comment]
> Thanks for coming back! I did this already and put my version 0.5.4 including some OpenCore add one into it which will only be used with iMac 2009-2011. Everything else remains the same except I added the Night Shift patch and an HDMI audio patch for MacPro3,1. There are only three forks off your current project. You will find it easily :-)

I added some of your fixes in the new version. The old shell scripts are kind of buggy in my opinion. Will be problem-free in the next release.

**Comment by Ausdauersportler** [comment]
> I added some of your fixes in the new version. The old shell scripts are kind of buggy in my opinion. Will be problem-free in the next release.

If you want I can help with those scripts. Have some experience with shell scrips collected over the years. As of now it should work, but I did not want to completely rewrite them because I did not spent enough time into checking out each line of code. 

Just made the adjustments to make it more compatible with the latest micropatcher versions. Barry already laid everything out to have a auto detect of models. One can work on this to auto detect more hardware, but this would only apply to those systems which can be hardware modified, so iMacs until 2011, all MacPro, some MacBookPro (new WiFi/BT modules).
@dosdude1 relied completely on the fixed hardware configurations Apple sold over the years. So his patcher was designed to install patches not needed on upgrades hardware (although I have to agree that this makes possibly only 1% of all cases).

But now we have an entirely new situation and only two groups of systems left supportable by a Big Sur patcher:
- Big Sur runs on 2012+ hardware mostly unpatched (you need the patcher only to boot the installer)
- Big Sur runs on hardware modified to become compatible, i.e. metal GPU and WiFi upgrades
The second group needs auto detection and for that reason I would spend some more work on it.

**Comment by minh-ton** [comment]
Yeah. The `postautomator.sh` script needs to be more polished. I think some more patches (NightShift...) should be added to the script for the fullest experience.

**Comment by Ausdauersportler** [comment]
Currently I have added the Nigh Shift patch by default with 0.5.4, so using my fork does it install automatically **if and only if** a system will be patched at all. 

Most 2012+ system does not need a single patch and the micropatcher exits gracefully if recognising such a system. Since OC offers OTA delta upgrades for unpatched systems only I would leave it that way. OTA upgrades outweigh the Night Shift "nice to have" feature by far.

I looks a little bit as OCLP will be the choice for 2012+ systems while micropatcher and real patching will remain to close the gap for pre 2011 system.  Currently I cannot get the needed extensions injected with OpenCore and from what I read on the OpenCore bug tracker it might not be possible in the way we did this before with Big Sur.

**Comment by moosethegoose2213** [comment]
> Yeah. The `postautomator.sh` script needs to be more polished. I think some more patches (NightShift...) should be added to the script for the fullest experience.

Night Shift would be really cool, and I agree with the postautomator. I plan to work on that some in the future when I have a chance. I'm sorry for not responding to these, I didn't get a notification. Sorry!

**Comment by minh-ton** [comment]
> > Yeah. The `postautomator.sh` script needs to be more polished. I think some more patches (NightShift...) should be added to the script for the fullest experience.
> 
> Night Shift would be really cool, and I agree with the postautomator. I plan to work on that some in the future when I have a chance. I'm sorry for not responding to these, I didn't get a notification. Sorry!

We've finished that script 😅
